Add http head requests (#673)
dictu-lang/Dictu: docs/docs/standard-lib/http.md

httpClient.put("https://httpbin.org/put", {"test": 10});
```

### httpClient.head(String) -> Result\<Response>

Sends a HTTP HEAD request to a given URL.
Returns a Result and unwraps to a Response upon success.

```cs
httpClient.head("https://httpbin.org/get");

{"content": "", "headers": ["...", "..."], "statusCode": 200}
```

### Response

> Both HTTP.get(), HTTP.post(), httpClient.put(), and httpClient.head() return a Result that unwraps a Response object on success, or nil on error.

Reviewer: Probably worth just stating all of them return the response

```suggestion
All HTTP requests return a Result that unwraps a Response object on success, or nil on error.
```
